### PR TITLE
Register webhook GVKs for validating callbacks

### PR DESCRIPTION
**Pull request by pierDipi**
When a callback is registered for a gvk that isn't part of the handlers that callback is never called since there is no webhook rule associated with that GVK.

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

# Changes

- Register webhook GVKs for validating callbacks
- Add unit test

/kind bug

Fixes #

**Release Note**

```release-note
Register webhook GVKs for validating callbacks
```

### CODE REVIEW

File: webhook/resourcesemantics/validation/reconcile_config.go [hunk 1]
```diff
@@ -143,6 +144,29 @@ func (ac *reconciler) reconcileValidatingWebhook(ctx context.Context, caCert []b
 			},
 		})
 	}
+	for gvk, callback := range ac.callbacks {
```
Comment (matzew) [inline webhook/resourcesemantics/validation/reconcile_config.go:147]: Is this related to the recent broker webhook issue that @aliok ran into?

Comment (pierDipi) [reply webhook/resourcesemantics/validation/reconcile_config.go:147]: Yes

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by aliok** [comment]
This is not changing the API of registering the callbacks and creating webhooks right?

File: webhook/resourcesemantics/validation/reconcile_config_test.go [hunk 2]
```diff
@@ -79,6 +80,13 @@ func TestReconcile(t *testing.T) {
 
 	// These are the rules we expect given the context of "handlers".
 	expectedRules := []admissionregistrationv1.RuleWithOperations{{
+		Operations: []admissionregistrationv1.OperationType{"CREATE", "DELETE", "UPDATE"},
+		Rule: admissionregistrationv1.Rule{
+			APIGroups:   []string{"pkg.knative.dev"},
+			APIVersions: []string{"v1alpha1"},
+			Resources:   []string{"callbackresources", "callbackresources/status"},
+		},
+	}, {
```
Comment (aliok) [inline webhook/resourcesemantics/validation/reconcile_config_test.go:89]: What is this `callbackresource` exactly?

**Comment by pierDipi** [comment]
No, it's not

**Comment by aliok** [comment]
Now that I've inspected the code more. I like this change. Thanks for fixing.

I was thinking problem here was more complicated but it turns out it is a small bug.

**Pull request merged**